contracts/RewardController.sol

```diff
@@ -0,0 +1,276 @@
+            reward =
+            reward + getRewardForBlocksRange(_fromBlock,
+                                            nextUpdateBlock,
+                                            vaultAllocPoint,
+                                            globalVaultsUpdates[i].totalAllocPoint);
+            _fromBlock = nextUpdateBlock;
+        }
+        return reward + getRewardForBlocksRange(_fromBlock,
+                                                block.number,
+                                                vaultAllocPoint,
+                                                globalVaultsUpdates[i].totalAllocPoint);
+    }
+
+    function getRewardForBlocksRange(uint256 _fromBlock, uint256 _toBlock, uint256 _allocPoint, uint256 _totalAllocPoint)
+    public
+    view
+    returns (uint256 reward) {
+        if ((_fromBlock >= startBlock && _toBlock >= _fromBlock) && _totalAllocPoint > 0) {
+            uint256 result;
+            uint256 i = (_fromBlock - startBlock) / epochLength + 1;
+            for (; i <= NUMBER_OF_EPOCHS; i++) {
+                uint256 endBlock = epochLength * i + startBlock;
+                if (_toBlock <= endBlock) {
+                    break;
+                }
+                result += (endBlock - _fromBlock) * epochRewardPerBlock[i-1];
+                _fromBlock = endBlock;
+            }
+            result += (_toBlock - _fromBlock) * (i > NUMBER_OF_EPOCHS ? 0 : epochRewardPerBlock[i-1]);
+            reward = result * _allocPoint / _totalAllocPoint;
+        }
+    }
+
+    /**
+    * @notice calculate the amount of rewards an account can claim for having contributed to a specific vault
+    * @param _vault the vault address
+    * @param _user the account for which the reward is calculated
+    */
+    function getPendingReward(address _vault, address _user) external view returns (uint256) {
+        if (IHATVault(_vault).rewardControllerRemoved(address(this))) {
+            return unclaimedReward[_vault][_user];
+        }
+        VaultInfo memory vault = vaultInfo[_vault];
+        uint256 rewardPerShare = vault.rewardPerShare;
+        uint256 totalShares = IERC20Upgradeable(_vault).totalSupply();
+
+        if (vault.lastRewardBlock != 0 && block.number > vault.lastRewardBlock && totalShares > 0) {
+            uint256 reward = getVaultReward(_vault, vault.lastRewardBlock);
+            rewardPerShare += (reward * REWARD_PRECISION / totalShares);
+        }
+
+        return IERC20Upgradeable(_vault).balanceOf(_user) * rewardPerShare / REWARD_PRECISION + 
+                unclaimedReward[_vault][_user] -
+                rewardDebt[_vault][_user];
+    }
+
+    /**
+    * @notice Transfer any tokens held in this contract to the owner
+    * @param _token the token to sweep
+    * @param _amount the amount of token to sweep
+    */
+    function sweepToken(IERC20Upgradeable _token, uint256 _amount) external onlyOwner {
+        _token.safeTransfer(msg.sender, _amount);
```

Comment:
why to pass the token and not use the `rewardToken` ?

Reply:
perhaps other tokens were sent to the contract by mistake

Reply:
so it covers that one more use case for free